PEP 587: version 5

vstinner opened the pull request.

Yhg1s commented on pep-0587.rst:
```diff
@@ -121,19 +121,19 @@ New functions:
-* ``PyInitError_Ok()``
+* ``PyInitStatus_Error(err_msg)``
+* ``PyInitStatus_Exit(exitcode)``
+* ``PyInitStatus_IsDone(status)``
```
I don't like the 'IsDone' name, because it's not clear that this means "Python can't continue" rather than "the initialisation is done". Perhaps "IsOk"? We can bikeshed on the name after b1, though.

vstinner replied:
It spent a significant time just to try to find a better name for this "error or exit" function! In Python, it's called "an exception".

Yhg1s commented on pep-0587.rst:
```diff
@@ -107,7 +107,7 @@ macros.
 
 * ``PyConfig``
-* ``PyInitError``
+* ``PyInitStatus``
```
Yes, this is a *much* better name :)

vstinner replied:
FYI the implementation of the PEP 587 has a long history: see "Issues of the PEP implementation" https://www.python.org/dev/peps/pep-0587/#python-issues I only added exitcode late in the development, when I finished to rewrite coreconfig.c/preconfig.c and when I started to work on main.c for real. So yeah, PyInitError is **no longer** a good name :-)

Two people (you and someone else, sorry I forgot) asked if we could drop "Init" from its name... Honestly, I'm not sure. **Maybe** it would make sense to reuse this structure for PyRun_SimpleXXX() functions, and in that case "PyInitStatus" would be surprising.

I'm not fully comfortable with "PyStatus" name, but I'm open to this idea :-)

vstinner commented:
@Yhg1s: I renamed PyInitStatus_IsDone() to PyInitStatus_Exception() and I renamed Py_ExitInitError() to Py_ExitInitException(). Are you ok with these names?

Second question: do you prefer to rename "PyInitStatus" to "PyStatus" to maybe use it later for PyRun_SimpleXXX() functions? Do are you ok with "PyInitStatus"?

Yhg1s commented:
The new names are a bit better, but I'm not sure they pop. I'm not particularly worried about these names, though.

I do like PyStatus better. I'm okay with the old names too, though. And we can bikeshed for weeks after it's in ;-P

vstinner commented:
I renamed PyInitStatus to PyStatus and I merged the version 5: https://github.com/python/peps/commit/fcdd72d719ecde5c161905a87eff81924f88f720
